tasks/borunov_v_block_partitioning/tests/functional/main.cpp

```diff
@@ -0,0 +1,125 @@
+#include <gtest/gtest.h>
+
+#include <algorithm>
+#include <array>
+#include <cmath>
+#include <cstddef>
+#include <string>
+#include <tuple>
+
+#include "borunov_v_block_partitioning/common/include/common.hpp"
+#include "borunov_v_block_partitioning/mpi/include/ops_mpi.hpp"
+#include "borunov_v_block_partitioning/seq/include/ops_seq.hpp"
+#include "util/include/func_test_util.hpp"
+
+namespace borunov_v_block_partitioning {
+
+class BorunovLinearFilterTest : public ppc::util::BaseRunFuncTests<InType, OutType, TestType> {
+ public:
+  static std::string PrintTestParam(const testing::TestParamInfo<typename BorunovLinearFilterTest::ParamType> &info) {
+    std::string task_name = std::get<1>(info.param);
+    auto test_params = std::get<2>(info.param);
+
+    int width = std::get<0>(test_params);
+    int height = std::get<1>(test_params);
+
+    std::string name = task_name + "_" + std::to_string(width) + "x" + std::to_string(height);
+
+    std::ranges::replace(name, ':', '_');
+    std::ranges::replace(name, '.', '_');
+    std::ranges::replace(name, '/', '_');
+
+    return name;
+  }
+
+ protected:
+  void SetUp() override {
+    TestType params = std::get<2>(GetParam());
+    int width = std::get<0>(params);
+    int height = std::get<1>(params);
+
+    const int fill_value = 42;
+
+    const std::size_t pixels = static_cast<std::size_t>(width) * static_cast<std::size_t>(height);
+    input_data_.resize(static_cast<std::size_t>(2) + pixels);
+    input_data_[0] = width;
+    input_data_[1] = height;
+
+    for (int i = 0; i < width * height; ++i) {
+      input_data_[2 + i] = fill_value;
+    }
```

Comment:
Does this filling actually test anything? Functional tests should test different cases, not the most trivial ones